### PR TITLE
Feat/create and remove asset

### PR DESCRIPTION
**Pull request by AinaVendrell**
#1574 does some modifications on new createAsset function

## Description
Create `createAsset` and `removeAsset` functions

## Approach taken / Explain the design
`createAsset` will return the id of the new asset and its url

## Testing

The pull request...

- has unit tests

### CODE REVIEW

File: packages/botonic-plugin-contentful/src/contentful/manage/manage-asset.ts [hunk 1]
```diff
@@ -51,6 +51,51 @@ export class ManageContentfulAsset {
...
   }
 
+  async removeAsset(context: ManageContext, assetId: AssetId): Promise<void> {
+    const environment = await this.environment
+    try {
+      const asset = await environment.getAsset(assetId.id)
+      if (asset.isPublished()) await asset.unpublish()
+      await asset.delete()
+    } catch (e) {
+      throw new CmsException('ERROR while deleting asset', e)
```
Comment (vanbasten17) [inline packages/botonic-plugin-contentful/src/contentful/manage/manage-asset.ts:61]: when giving information about an error, it's important to have a quick way to identify the resource in question. I would return `ERROR while deleting asset with ID: '${assetId}' .`

File: packages/botonic-plugin-contentful/src/contentful/manage/manage-asset.ts [hunk 2]
```diff
@@ -51,6 +51,51 @@ export class ManageContentfulAsset {
...
+    }
+  }
+
+  async createAsset(
+    context: ManageContext,
+    title: string,
+    fileName: string,
+    contentType: string,
+    file: any,
+    description?: string
```
Comment (vanbasten17) [inline packages/botonic-plugin-contentful/src/contentful/manage/manage-asset.ts:71]: having lots of parameters in a function can be a sign of having something hard to extend and maintain. I would do the following here:
* create an interface representing the asset you want to add
```
inferface NewAsset(AssetToCreate, ...) {
title: string,
fileName: string,
contentType: string,
file: any,
description?: string
}
```
* calling the function `createAsset` in this way `createAsset(context, {title: 'newTitle', fileName: 'fileName', ...}`

what do you think? :)

File: packages/botonic-plugin-contentful/src/contentful/manage/manage-contentful.ts [hunk 3]
```diff
@@ -102,4 +102,26 @@ export class ManageContentful implements ManageCms {
...
     return this.manageAsset.copyAssetFile(context, assetId, fromLocale)
   }
+
+  async createAsset(
+    context: ManageContext,
+    title: string,
+    fileName: string,
+    contentType: string,
+    file: any,
+    description?: string
```
Comment (vanbasten17) [inline packages/botonic-plugin-contentful/src/contentful/manage/manage-contentful.ts:112]: take a look at previous comment

**Review by vanbasten17 — CHANGES_REQUESTED** [review]
I think that you can rethink what I was suggesting to improve this, after that, you can go for it! Nice job!! 💯

**Comment by vanbasten17** [comment]
now I see that @dpinol has opened a new branch to solve this!

**Review by dpinol — APPROVED** [review]

File: packages/botonic-plugin-contentful/src/contentful/manage/manage-asset.ts (hunk 1 above)
Comment (dpinol) [reply packages/botonic-plugin-contentful/src/contentful/manage/manage-asset.ts:61]: @AinaVendrell here you can do new CmsException('ERROR while deleting asset', e, assetId)

**Pull request merged**